tests/compiler/function-array.ts

```diff
@@ -0,0 +1,24 @@
+function funcA(x: i32): i32 {
+  return x * 3;
+}
+
+function funcB(x: i32): i32 {
+  return x + 1;
+}
+
+function funcC(x: i32): i32 {
+  return x / 2;
+}
+
+const funcD = (x: i32): i32 => x % 4 + 5;
+const funcE = (x: i32): i32 => x - 7;
+
+const funcs = [funcA, funcB, funcC, funcD, funcE];
+const results = [369, 124, 61, 8, 116];
+
+for (let i = 0; i < results.length; i++) {
+  const expected = results[i];
+  const actual = funcs[i](123);
+
+  assert(expected == actual);
+}
```

Comment:
Could you simplify this example to a minimal example? Also, I guess make sense put in this test: https://github.com/AssemblyScript/assemblyscript/blob/main/tests/compiler/std/array-access.ts

Reply:
See the current change.